fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorBase.java

```diff
@@ -0,0 +1,156 @@
+import com.ibm.fhir.model.resource.Bundle.Entry;
+import com.ibm.fhir.model.type.Uri;
+import com.ibm.fhir.model.util.ModelSupport;
+import com.ibm.fhir.model.resource.OperationOutcome;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.server.operation.spi.FHIRResourceHelpers;
+import com.ibm.fhir.server.operation.spi.FHIRRestOperationResponse;
+
+/**
+ * Abstract base class of the {@link FHIRRestInteractionVisitor}. Manages access to the
+ * map managing local references (localRefMap) and the array of response Entry objects 
+ * (responseBundleEntries).
+ */
+public abstract class FHIRRestInteractionVisitorBase implements FHIRRestInteractionVisitor {
+    private static final Logger log = Logger.getLogger(FHIRRestInteractionVisitorBase.class.getName());
+    
+    protected static final com.ibm.fhir.model.type.String SC_BAD_REQUEST_STRING = string(Integer.toString(SC_BAD_REQUEST));
+    protected static final com.ibm.fhir.model.type.String SC_GONE_STRING = string(Integer.toString(SC_GONE));
+    protected static final com.ibm.fhir.model.type.String SC_NOT_FOUND_STRING = string(Integer.toString(SC_NOT_FOUND));
+    protected static final com.ibm.fhir.model.type.String SC_ACCEPTED_STRING = string(Integer.toString(SC_ACCEPTED));
+    protected static final com.ibm.fhir.model.type.String SC_OK_STRING = string(Integer.toString(SC_OK));
+
+    // the helper we use to do most of the heavy lifting
+    protected final FHIRResourceHelpers helpers;
+    
+    // Used to resolve local references
+    protected final Map<String, String> localRefMap;
+    
+    // Held as an array, so we can be sure of O(1) operations because entries are not processed in order
+    private final Entry[] responseBundleEntries;
+
+    /**
+     * Protected constructor
+     * @param helpers
+     * @param localRefMap
+     * @param responseBundleEntries
+     */
+    protected FHIRRestInteractionVisitorBase(FHIRResourceHelpers helpers,
+        Map<String, String> localRefMap, Entry[] responseBundleEntries) {
+        this.helpers = helpers;
+        this.localRefMap = localRefMap;
+        this.responseBundleEntries = responseBundleEntries;
+    }
+
+    /**
+     * Set the given entry e in the response bundle and log a bundle entry completion
+     * message.
+     * @param entryIndex
+     * @param e
+     * @param requestDescription
+     * @param initialTime
+     */
+    protected void setEntryComplete(int entryIndex, Entry e, String requestDescription, long initialTime) {
+        responseBundleEntries[entryIndex] = e;
+        logBundledRequestCompletedMsg(requestDescription, initialTime, e.getResponse().getStatus().getValue());
+    }
+
+    protected Entry getResponseEntry(int entryIndex) {
+        return responseBundleEntries[entryIndex];
+    }
+    
+    private void logBundledRequestCompletedMsg(String requestDescription, long initialTime, String httpStatus) {
+        StringBuffer statusMsg = new StringBuffer();
```

Comment:
the StringBuffer should be building what's in the info msg, not just the statusMsg